framework/doc/content/syntax/Materials/index.md

```diff
@@ -1,4 +1,12 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
+# Materials System Overview
+
+Time derivative of a material property could be needed for transient calculations.
+MOOSE provides a function getMaterialPropertyDot that can be called in kernels, bcs, user objects, etc,
+which returns a reference of a time derivative of a real scalar material property.
+The time derivative obtained with this function is consistent with the time integration scheme used by the calculation.
+It is noted that calling this function will make the material property stateful.
+MOOSE will keep three copies of the property on all quadrature points for the current, old and older
+properties that are used for evaluating the time derivative.
```

Comment:
Please add an even larger "note" about it being for scalar (`Real`) material properties only.  Use one of these: http://mooseframework.org/utilities/MooseDocs/extensions/alert.html